core/src/main/scala/org/bitcoins/core/wallet/fee/FeeUnit.scala

```diff
@@ -9,8 +9,14 @@ import org.bitcoins.core.protocol.transaction.Transaction
   */
 sealed abstract class FeeUnit {
   def currencyUnit: CurrencyUnit
+  def *(cu: CurrencyUnit): CurrencyUnit = this * cu.satoshis.toLong
+  def *(int: Int): CurrencyUnit = this * int.toLong
+  def *(long: Long): CurrencyUnit = currencyUnit * long
```

Comment:
This method absolutely needs a detailed scaladoc because it is the only non-abstract method that needs to be overridden sometimes. You have to make sure its super clear that people only override this method and not others. Also make clear what the responsibility is here wrt units.

Reply:
This seems like a big footgun, we should probably leave this as abstract in the `FeeUnit` and implement in our subtypes as I know I probably won't remember to override this next time we need to modify this